src/frontend/reducer/NotesReducer.js

```diff
@@ -3,6 +3,8 @@ const NotesInitialState = {
   notes: [],
   archives: [],
   trash: [],
+  isModalOpen: false,
+  noteBeingEdited: {},
```

Comment:
Nice! you can add the flag to see editing or not as well as the id being edited in one place! Smart.